### PR TITLE
Add session garbage collector check

### PR DESCRIPTION
**Pull request by Ruslan-Aleev**
### What does it do?
It often happens that the `modx_session` table grows uncontrollably (I have seen up to 5 gigabytes), which sometimes leads to complete non-operation of both the site and the server.

I added session garbage collector check in installer.

gc_installer_warn

### Why is it needed?
That there were no problems with overflow of the table `modx_session`.

### How to test
Make the installation from scratch by changing the `session.gc_probability, session.gc_divisor` parameters.

### Related issue(s)/PR(s)
https://github.com/modxcms/revolution/pull/16287
https://github.com/modxcms/revolution/issues/16275

### CODE REVIEW

**Review by Mark-H — DISMISSED** [review]

File: setup/includes/test/modinstalltest.class.php [hunk 1]
```diff
@@ -555,6 +556,30 @@ public function _checkSuhosin()
         $this->install->settings->store();
     }
 
+    /**
+     * Check sessions garbage collector
+     */
+    protected function _checkSessionsGarbageCollector()
+    {
+        $success = false;
```
Comment (theboxer) [inline setup/includes/test/modinstalltest.class.php]:
```suggestion
        $status = 'success';
```

File: setup/includes/test/modinstalltest.class.php [hunk 2]
```diff
@@ -555,6 +556,30 @@ public function _checkSuhosin()
...
+        $gc_probability = (int)@ini_get('session.gc_probability');
+        $gc_divisor = (int)@ini_get('session.gc_divisor');
+
+        if (!$gc_probability) {
+            $success = @ini_set('session.gc_probability', 1);
+            $success = $success !== false ? true : false;
+        } else {
+            $success = true;
+        }
```
Comment (theboxer) [inline setup/includes/test/modinstalltest.class.php]:
```suggestion
        }
```

File: setup/includes/test/modinstalltest.class.php [hunk 3]
```diff
@@ -555,6 +556,30 @@ public function _checkSuhosin()
...
+     */
+    protected function _checkSessionsGarbageCollector()
+    {
+        $success = false;
+        $gc_probability = (int)@ini_get('session.gc_probability');
+        $gc_divisor = (int)@ini_get('session.gc_divisor');
+
+        if (!$gc_probability) {
+            $success = @ini_set('session.gc_probability', 1);
+            $success = $success !== false ? true : false;
```
Comment (theboxer) [inline setup/includes/test/modinstalltest.class.php]:
```suggestion
            $status = @ini_set('session.gc_probability', 1) !== false ? 'success' : 'fail';
```

File: setup/includes/test/modinstalltest.class.php [hunk 4]
```diff
@@ -555,6 +556,30 @@ public function _checkSuhosin()
...
+        } else {
+            $success = true;
+        }
+
+        $this->title('session_gc', $this->install->lexicon('test_session_gc'));
+        if ($success) {
+            $this->warn('session_gc', '', $this->install->lexicon('test_session_gc_success', ['gc_probability' => $gc_probability, 'gc_divisor' => $gc_divisor]));
+        } else {
+            $this->warn('session_gc', '', $this->install->lexicon('test_session_gc_fail', ['gc_probability' => $gc_probability, 'gc_divisor' => $gc_divisor]));
+        }
```
Comment (theboxer) [inline setup/includes/test/modinstalltest.class.php]:
```suggestion
        $this->warn('session_gc', '', $this->install->lexicon("test_session_gc_$status", ['gc_probability' => $gc_probability, 'gc_divisor' => $gc_divisor]));
        
```

**Review by theboxer — CHANGES_REQUESTED** [review]
I'd suggest following changes, it removes some unnecessary ifs and makes it slightly more readable.

**Review by theboxer — APPROVED** [review]
👍

**Comment by Mark-H** [comment]
Do the lexicons need to be in both default.inc.php and test.inc.php?

**Comment by Ruslan-Aleev** [comment]
@Mark-H To be honest, I don’t know, I haven’t figured it out. Other keys are present in both files.

**Pull request merged**